Fix HTCondor format for trailing empty fields

maxfischer2781 opened the pull request:
This PR adjusts the `htcondor` parser to match its own example and HTCondor-CE. It is still compatible with the old format.

- HTCondor parser now ignores a trailing, *empty* field after all required ones. This means the final `|` can act both as a delimiter and terminator.
- The example is adjusted for consistency: The query now has a newline to make it feasible for multiple jobs, and the output has the trailing `|` of the query.

Closes #341.

maxfischer2781 committed:
ignore empty trailing fields

maxfischer2781 committed:
update example for multiple jobs

maxfischer2781 committed:
Example output matches query

maxfischer2781 commented:
Ugh, the Codacy lint now complains about the line length but that was already too long before. Breaking up the line to fit in the length seems counterproductive, since reproducing the actual format is kind of the point here.

What's the policy here? Should I break up the line regardless? If I'm reading the Codacy report right, this is just prettified Pylint report - is it okay to [disable the inspection for those lines or the entire block](https://pylint.readthedocs.io/en/latest/user_guide/messages/message_control.html#block-disables)?

tofu-rocketry commented:
Don't worry about the line length. I need to sort the code style checks to have a more generous limit, but that requires a general overhaul of the CI tools.

maxfischer2781 commented:
@tofu-rocketry Thanks for updating the style checks, this is ready for review.

tofu-rocketry reviewed (changes requested):
Oops, realised I should update the unit tests. Can sort that myself.

tofu-rocketry committed:
Add test case for HTCondor parsing

A possible out from HTCondor-CE is a trailing pipe symbol and no scaling
factor (as it uses the pipe as a value terminator. Parser would split
this and interpret the scale factor as an empty string which would make
the parsing fail. This adds a test case to check this situation.

gregcorbett reviewed (approved):
LGTM

tofu-rocketry commented:
Thanks, @gregcorbett
